x/params/keeper/authority.go

```diff
@@ -0,0 +1,21 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/strangelove-ventures/paramauthority/x/params/types"
+	"github.com/strangelove-ventures/paramauthority/x/params/types/proposal"
+)
+
+// SetParams sets the total set of params parameters.
+func (k Keeper) SetParams(ctx sdk.Context, params proposal.Params) {
+	k.paramSpace.SetParamSet(ctx, &params)
+}
+
+// GetAuthority returns the module authority
+func (k Keeper) GetAuthority(
+	ctx sdk.Context,
+) (authority string) {
+	var res string
+	k.paramSpace.Get(ctx, []byte(types.AuthorityKey), &res)
```

Comment:
please use storey here. this subspace is mainlkly for params in json encoding

Reply:
Without throwing it in the param space, I couldn't find a great way to use the `MsgUpdateParams` to update the `authority` without introducing another msg type. Any ideas here?

Reply:
how come the normal store doesn't work?

Reply:
e2e test [here](https://github.com/strangelove-ventures/noble/blob/main/ibctest/paramauthority_test.go)

https://github.com/strangelove-ventures/noble/blob/main/ibctest/paramauthority_test.go#L44-L46
AFAICS an update params proposal and, by extension, this `MsgUpdateParams` following the same implementation, it's only possible to update params within a param subspace, not directly in a store.

Reply:
its either or, but forgot this was a param, then its chill